image_tf_encoder.py

```diff
@@ -0,0 +1,119 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Union, Iterable, List, Any, Optional, Dict
+
+import numpy as np
+from jina import DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+
+
+def _batch_generator(data: List[Any], batch_size: Optional[int]):
+    for i in range(0, len(data), batch_size):
+        yield data[i:i + batch_size]
+
+
+class ImageTFEncoder(Executor):
+    """
+    :class:`ImageKerasEncoder` encodes ``Document`` content from a ndarray,
+    potentially B x (Channel x Height x Width) into a ndarray of `B x D`.
```

Comment:
In order to unify the image input shapes among all encoders, we assume (h, w, 3).
In case preprocessing is done inside the encoder, we assume `dtype=uint`. Otherwise `dtype=float32`.